packages/cli/test/composites.test.ts

```diff
@@ -24,9 +37,222 @@ describe('composites', () => {
+      expect(Object.keys(encodedComposite.models).length).toEqual(3)
+      const ceramicClient = new CeramicClient()
+      const composite = await Composite.fromJSON({
+        ceramic: ceramicClient,
+        definition: encodedComposite,
+      })
+      const modelNames = Object.values(composite.toParams().definition.models).map((modelDefinition) => {
+          return modelDefinition.name
+        }
+      )
+      const extractModel = await execa('glaze', [
+        'composite:extract-model',
+        'test/mocks/encoded.composite.profiles.json',
+        modelNames[0],
+        modelNames[1],
+      ])
+      const newEncodedComposite = JSON.parse(
+        extractModel.stdout.toString()
+      ) as EncodedCompositeDefinition
+      const newComposite = await Composite.fromJSON({
+        ceramic: ceramicClient,
+        definition: newEncodedComposite,
+      })
+      expect(Object.keys(newEncodedComposite.models).length).toEqual(2)
+      const newModelNames = Object.values(newComposite.toParams().definition.models).map((modelDefinition) => {
+          return modelDefinition.name
+      })
+      expect(newModelNames.length).toEqual(2)
+      expect(newModelNames.includes(modelNames[0])).toBeTruthy()
+      expect(newModelNames.includes(modelNames[1])).toBeTruthy()
+    }, 60000)
+  })
+
+  describe('composite:compile', () => {
+    test('composite compilation fails without the composite path and at least one output path param', async () => {
+      const compileWithNoParams = await execa('glaze', ['composite:compile'])
+      expect(
+        compileWithNoParams.stderr
+          .toString()
+          .includes('Missing composite path and at output path')
+      ).toBe(true)
+
+      const compileWithJustCompositePath = await execa('glaze', [
+        'composite:compile',
+        'test/mocks/encoded.composite.profiles.json',
+      ])
+      expect(
+        compileWithJustCompositePath.stderr
+          .toString()
+          .includes('Missing composite path and at output path')
+      ).toBe(true)
+    }, 60000)
+
+    test('composite compilation succeeds', async () => {
+      const dirpath = 'test/test_output_files'
+      const filename = 'runtime.composite.profiles'
+      const compileWithJustCompositePath = await execa('glaze', [
+        'composite:compile',
+        'test/mocks/encoded.composite.profiles.json',
+        `${dirpath}/${filename}.json`,
+        `${dirpath}/${filename}.js`,
+        `${dirpath}/${filename}.ts`,
+      ])
+      expect(
+        compileWithJustCompositePath.stderr
+          .toString()
+          .includes('Successfully saved compiled composite into given path(s)')
+      ).toBe(true)
+
+      const jsonRepresentation = await fs.readFile(`${dirpath}/${filename}.json`)
+      const jsRepresentation = await fs.readFile(`${dirpath}/${filename}.js`)
+      const tsRepresentation = await fs.readFile(`${dirpath}/${filename}.ts`)
```

Comment:
All these files could be read in parallel:
```suggestion
      const [jsonRepresentation, jsRepresentation, tsRepresentation] = await Promise.all([
        fs.readFile(`${dirpath}/${filename}.json`),
        fs.readFile(`${dirpath}/${filename}.js`),
        fs.readFile(`${dirpath}/${filename}.ts`),
      ])
```